libs/ngrx-toolkit/src/lib/mutation/rx-mutation.spec.ts

```diff
@@ -0,0 +1,594 @@
+import { delay, Observable, of, Subject, switchMap, throwError } from 'rxjs';
+import { concatOp, exhaustOp, mergeOp, switchOp } from '../flattening-operator';
+import { rxMutation } from './rx-mutation';
+
+type Param =
+  | number
+  | {
+      value: number | Observable<number>;
+      delay?: number;
+      fail?: boolean;
+    };
+
+type NormalizedParam = {
+  value: number | Observable<number>;
+  delay: number;
+  fail: boolean;
+};
+
+async function asyncTick(): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, 0);
+  });
+}
+
+function calcDouble(value: number, delayInMsec = 1000): Observable<number> {
+  return of(value * 2).pipe(delay(delayInMsec));
+}
+
+function fail(_value: number, delayInMsec = 1000): Observable<number> {
+  return of(null).pipe(
+    delay(delayInMsec),
+    switchMap(() => throwError(() => ({ error: 'Test-Error' }))),
+  );
+}
+
+function createTestSetup(flatteningOperator = concatOp) {
+  function normalizeParam(param: Param): NormalizedParam {
+    if (typeof param === 'number') {
+      return {
+        value: param,
+        delay: 1000,
+        fail: false,
+      };
+    }
+
+    return {
+      value: param.value,
+      delay: param.delay ?? 1000,
+      fail: param.fail ?? false,
+    };
+  }
+
+  type SuccessParams = { result: number; params: Param };
+  type ErrorParams = { error: unknown; params: Param };
+
+  let onSuccessCalls = 0;
+  let onErrorCalls = 0;
+  let counter = 3;
+
+  let lastOnSuccessParams: SuccessParams | undefined = undefined;
+  let lastOnErrorParams: ErrorParams | undefined = undefined;
+
+  return TestBed.runInInjectionContext(() => {
+    const increment = rxMutation({
+      operation: (param: Param) => {
+        const normalized = normalizeParam(param);
+
+        if (normalized.value instanceof Observable) {
+          return normalized.value.pipe(
+            switchMap((value) => {
+              if (normalized.fail) {
+                return fail(value, normalized.delay);
+              }
+              return calcDouble(value, normalized.delay);
+            }),
+          );
+        }
+
+        if (normalized.fail) {
+          return fail(normalized.value, normalized.delay);
+        }
+        return calcDouble(normalized.value, normalized.delay);
+      },
+      operator: flatteningOperator,
+      onSuccess: (result, params) => {
+        lastOnSuccessParams = { result, params };
+        onSuccessCalls++;
+        counter = counter + result;
+      },
+      onError: (error, params) => {
+        lastOnErrorParams = { error, params };
+        onErrorCalls++;
+      },
+    });
+
+    return {
+      increment,
+      getCounter: () => counter,
+      onSuccessCalls: () => onSuccessCalls,
+      onErrorCalls: () => onErrorCalls,
+      lastOnSuccessParams: () => lastOnSuccessParams,
+      lastOnErrorParams: () => lastOnErrorParams,
+    };
+  });
+}
+
+describe('rxMutation', () => {
```

Comment:
Since we do a test refactoring, I'll leave it here as it is. I will make sure that we have a test coverage which covers both cases (yours and mine)